templates/components/common/cart-preview.html

```diff
@@ -1,100 +1,159 @@
-<div>
-    <div>
-        {{#if cart.items.length}}
-            <ul>
-                {{#each (limit cart.items 4)}}
-                    <li>
-                        <div>
+<div class="c-cart-preview__container">
+    {{#if cart.items.length}}
+        <ul class="c-cart-preview__list">
+            {{#each (limit cart.items 4)}}
+                <div class="js-item-row c-cart__item c-cart-preview__item" data-type-product-id="{{product_id}}">
+                    <div class="c-cart-preview__item-information">
+                        <div class="c-cart__item-image-wrapper">
                             {{#if type '==' 'GiftCertificate'}}
                                 <img
-                                    alt="GiftCertificate"
-                                    title="GiftCertificate"
                                     src="{{cdn ../theme_settings.default_image_gift_certificate}}"
+                                    alt="{{lang 'cart.gift_certificates.gift_certificate'}}"
+                                    title="{{lang 'cart.gift_certificates.gift_certificate'}}"
+                                    class="c-cart__item-image"
                                 >
                             {{else}}
-                                {{> components/common/image 
-                                    image=image 
-                                    fallback_size=../theme_settings.productthumb_size
+                                {{> components/common/image
+                                    image=image
+                                    fallback_size=../theme_settings.product_thumb_size
                                     default_image=../theme_settings.default_image_product
+                                    class="c-cart__item-image"
                                 }}
                             {{/if}}
                         </div>
-                        <div>
-                            <span>
-                                {{brand.name}}
-                            </span>
-
-                            <div>
-                                <a href="{{url}}" alt="{{name}}" title="{{name}}">
-                                    {{name}}
-                                </a>
+                        <div class="c-cart__item-head">
+                            <a class="o-text o-text--xl o-text--bold c-cart__item-title" href="{{url}}">
+                                {{name}}
+                            </a>
+                            <div class="c-cart__item-rating-wrapper">
+                                <div class="c-cart__item-rating" role="img" aria-label="">
+                                    <span class="c-cart-item__filled-stars">
+                                        {{> components/common/star status='filled' class="u-hidden"}}
```

Comment:
It looks like a lot of repetitive code. Also, shouldn't the star icon remain the same regardless of its full/empty state and just change its filling? 

Considering the logic, there's no way the product could have 3.5 rating?

Reply:
@aleksandraworhacz This was previously refactored to be like this in a previous CR. Remember I used to inject the svg HTML via JS? I know it looks repetitive and there is no way with this to achieve 3.5 rating in the current format. What do you think, how should we handle this so we can just work on it once and make it perfect without changing it again in the future? My only idea is to just do it with JavaScript, but maybe there is a better solution 🤔

Reply:
@mgbelegu I have in mind something like:

Single star code:
```
<div class="c-rating-star">
    <svg width="12" height="12" xmlns="http://www.w3.org/2000/svg" viewBox="-1 -1 14 14" class="c-rating-star__icon">
      <defs>
        <linearGradient id="star-part-{{id}}">
          <stop offset="0%" class="star-fill" />
          <stop offset="{{fill}}" class="star-fill" />
          <stop offset="{{fill}}" stop-color="white" />
          <stop offset="100%" stop-color="white" />
        </linearGradient>
      </defs>

      <path d="m6 0 1.67 4.22 4.33.37-3.3 2.97 1.01 4.44-3.71-2.38-3.71 2.38 1.01-4.44-3.3-2.98 4.33-.37z" fill="url(#star-part-{{id}})" class="c-rating-star__stroke" stroke-width="0.8" />
    </svg>
</div>
```
where `id` is a loop index and fill represents how much % of the star is supposed to be filled. 

and the `c-rating`component would be like:
```
<div class="c-rating">
    {{#for 1 5 }}
        [ still = check if value - $index > 1 -> if so set 100% else multiply value by 100% ]
        {{> components/rating/rating-star id="{{$index}}" fill="{{starFill}}%"}}
    {{/for}}
</div>

```
where the value is overall rating e.g. `3.5`


the styles would be smth like:
```
.c-rating-star {
  --icon-color: #c1a45a;

  display: inline-flex;
  font-size: 12px;
}

.c-rating-star__icon {
  width: 1em;
  height: 1em;
}

.star-fill {
  stop-color: var(--icon-color, #000);
}

.c-rating-star__stroke {
  stroke: var(--icon-color, #000);
}

```

I'm not sure tho if it's possible tho, I remember that math operations and variables in bigCommerce were really terrible in terms of what can you do :/ 

Or you could always use js to set proper `offset` values :) 

(sorry if the message was chaotic 🙈 )

Reply:
I've changed it. Can you please review? I think it works much better now.

Please note that I've done this on purpose:
`<div class="c-rating">
    {{> components/common/star index='1'}}
    {{> components/common/star index='2'}}
    {{> components/common/star index='3'}}
    {{> components/common/star index='4'}}
    {{> components/common/star index='5'}}
</div>`

because using the for loop for some reason it doesn't work. It gives me a partial issue saying the component cannot be found. If I add `{{#partial "page"}}` to the component the cart preview view doesn't work.